proto/viam/app/mltraining/v1/ml_training.proto

```diff
@@ -0,0 +1,68 @@
+syntax = "proto3";
+
+package viam.app.mltraining.v1;
+
+import "app/data/v1/data.proto";
+import "google/protobuf/struct.proto";
+import "google/protobuf/timestamp.proto";
+
+option go_package = "go.viam.com/api/app/mltraining/v1";
+
+service MLTrainingService {
+  rpc SubmitTrainingJob(SubmitTrainingJobRequest) returns (SubmitTrainingJobResponse);
+  rpc GetTrainingJob(GetTrainingJobRequest) returns (GetTrainingJobResponse);
+}
+
+enum ModelType {
+  MODEL_TYPE_UNSPECIFIED = 0;
+  MODEL_TYPE_MULTI_CLASS_CLASSIFICATION = 1;
+  MODEL_TYPE_MULTI_LABEL_CLASSIFICATION = 2;
+}
+
+message SubmitTrainingJobRequest {
+  viam.app.data.v1.Filter filter = 1;
+  string org_id = 2;
+  string model_name = 3;
+  string model_version = 4;
+  ModelType model_type = 5;
+  repeated string training_labels = 6;
+}
+
+message SubmitTrainingJobResponse {
+  string id = 1;
+}
+
+message GetTrainingJobRequest {
+  string id = 1;
+}
+
+message GetTrainingJobResponse {
+  string id = 1;
+  TrainingJobMetadata metadata = 2;
+}
+
+enum TrainingStatus {
+  TRAINING_STATUS_UNSPECIFIED = 0;
+  TRAINING_STATUS_PENDING = 1;
+  TRAINING_STATUS_IN_PROGRESS = 2;
+  TRAINING_STATUS_COMPLETED = 3;
+  TRAINING_STATUS_FAILED = 4;
+}
+
+message TrainingJobMetadata {
```

Comment:
Would ideally store the tags that we trained on, enabling the user to easily retrain on the same tags if retraining/updating an existing model.

Reply:
Those tags are in the `SubmitTrainingJobRequest request` field, so will still be saved/queryable